src/main/java/uk/gov/justice/hmpps/casenotes/repository/CaseNoteTypeRepository.kt

```diff
@@ -1,11 +1,19 @@
 package uk.gov.justice.hmpps.casenotes.repository
 
 import org.springframework.data.jpa.repository.JpaRepository
+import org.springframework.data.jpa.repository.Query
 import org.springframework.stereotype.Repository
 import uk.gov.justice.hmpps.casenotes.model.CaseNoteType
 import java.util.Optional
 
 @Repository
 interface CaseNoteTypeRepository : JpaRepository<CaseNoteType, Long> {
-  fun findCaseNoteTypeByParentTypeTypeAndType(parentType: String, type: String): Optional<CaseNoteType>
+  @Query(
+    """
+        select t from CaseNoteType t
+        join fetch t.parentType pt
+        where pt.type = :parentType and t.type = :type
+    """,
+  )
+  fun findByParentTypeAndType(parentType: String, type: String): Optional<CaseNoteType>
```

Comment:
Again more for my interest. Is the defined query here to reduce the function name or is this a more optimised query than was being generated?